docs/intelligentapps/agentbuilder.md

```diff
@@ -0,0 +1,158 @@
+- Generate starter prompts
+- Iterate and refine with each run
+- Break down complex tasks through prompt chaining and structured outputs
+- Provide easy access to code for seamless Large Language Model (LLM) integration via APIs
+
+Agent Builder also enhances intelligent app's capabilities with tool use:
+- Connect to existing MCP servers
+- Build a new MCP server from scaffold and test in Agent Builder
+
+![Getting started with prompt builder](./images/promptbuilder/AgentBuilder.gif)
+
+## Create, edit, and test prompts
+
+To access Agent Builder, use either of these options:
+
+- In the AI Toolkit view, select **Agent (Prompt) Builder**
+- Select **Try in Agent (Prompt) Builder** from a model card in the model catalog
+
+To test a prompt in Agent Builder, follow these steps:
+
+1. In **Models**, select a model from the dropdown list, or select **Browse models** to add another model from the model catalog.
+
+    ![select a model](./images/promptbuilder/select_models.png)
+
+1. Enter a **User prompt** and optionally enter a **System prompt**.
+
+   The *user prompt* is the input that you want to send to the model. The optional *system prompt* is used to provide instructions with relevant context to guide the model response.
+
+   > [!TIP]
+   > You can describe your project idea by using natural language and let the AI-powered feature generate the prompts for you to experiment with.
+   > ![generate prompts with natural language](./images/promptbuilder/generate_prompt_2.gif)
+
+1. Select **Run** to send the prompts to the selected model.
+
+1. Optionally, select **Add Prompt** to add more user and assistant prompts to the conversation, or select **Add to Prompts** as the history and context you send to the model to further guide the model's behavior.
+
+1. Repeat the previous steps to iterate over your prompts by observing the model response and making changes to the prompts.
+
+## Use MCP servers
+MCP server is a tool that allows you to connect to external APIs and services, enabling your agent to perform actions beyond just generating text. For example, you can use an MCP server to access databases, call web services, or interact with other applications.
+
+You can use the agent builder to discover and configure featured MCP servers, connect to existing MCP servers or build a new MCP server from scaffold.
+
+### Configure MCP server from featured selections
+AI Toolkit provides a list of featured MCP servers that you can use to connect to external APIs and services.
+
+To configure an MCP server from featured selections, follow these steps:
+1. In the **Tools** section, select **+ MCP Server** and select **+ Add server** in the quick pick.
+2. Select **Use Featured MCP Servers** from the dropdown list.
+3. Select a MCP server that fit your needs.
+![connect to MCP server](./images/promptbuilder/Featured_server.png)
+
+For servers that are ready to use, AI Toolkit will prompt you to select tools and add to your agent afterwards.
```

Comment:
Featured server may need `npx` or `uv` dependency, AITK will validate if `npx` or `uv` is installed, if not, AITK will prompt user that `npx` or `uv` is not installed. Can we add this information here? And also provide the installation link for them:
`node`: https://nodejs.org/en/download
`npx`: `npm install -g npx`
`uv`: https://docs.astral.sh/uv/getting-started/installation/

When this information is added here, I can add a `Learn more` link in the `npx` or `uv` not installed prompt.

Reply:
Updated